src/Mooncake.jl

```diff
@@ -142,33 +142,11 @@ include("interface.jl")
 include("config.jl")
 include("developer_tools.jl")
 
-export primal,
-    tangent,
-    randn_tangent,
-    increment!!,
-    NoTangent,
-    Tangent,
-    MutableTangent,
-    PossiblyUninitTangent,
-    set_to_zero!!,
-    tangent_type,
-    zero_tangent,
-    _scale,
-    _add_to_primal,
-    _diff,
-    _dot,
-    zero_codual,
-    codual_type,
-    rrule!!,
-    build_rrule,
-    value_and_gradient!!,
-    value_and_pullback!!,
-    NoFData,
-    NoRData,
-    fdata_type,
-    rdata_type,
-    fdata,
-    rdata,
-    get_interpreter
+# Public, not exported
+include("public.jl")
+@public value_and_pullback!!, prepare_pullback_cache
+
```

Comment:
```suggestion
@public tangent_type, zero_tangent
```
DI needs these two as well

Reply:
Why is it that DI needs `zero_tangent`? I know that it need `tangent_type` to check whether the user has provided the correct thing before calling `Mooncake.value_and_pullback!!`, but I'm not sure why we need `zero_tangent`.

Reply:
Hmmm my feeling is that we're currently doing this the wrong way in DI. I should really just make it possible to access the memory allocated by `prepare_pullback_cache` for the tangent of `y`, so that you don't have to call `zero_tangent`.

To be honest, I think I would like to make a breaking release of Mooncake in the next couple of days with refined definitions of `value_and_gradient!!` and `value_and_pullback!!`. Would you object to depending on internals for another couple of days?

Reply:
No worries! But maybe this shouldn't stop us from merging this PR?

Reply:
Ah, yes, sorry. My point was going to be that I would release this PR without making either of these functions public. Would that be acceptable?

Reply:
Definitely. Since Mooncake's interface was undefined until now, I've been relying on internals this whole time after all ^^